src/rulesets/Base/indexing.jl

```diff
@@ -1,3 +1,55 @@
+#####
+##### getindex(::Tuple)
+#####
+
+function frule((_, ẋ), ::typeof(getindex), x::Tuple, i::Integer)
+    return x[i], ẋ[i]
+end
+
+function frule((_, ẋ), ::typeof(getindex), x::Tuple, i)
+    y = x[i]
+    return y, Tangent{typeof(y)}(ẋ[i]...)
+end
+
+"for a given typle type, returns a Val{N} where N is the length of the tuple"
+_tuple_N(::Type{<:Tuple{Vararg{<:Any, N}}}) where {N} = Val(N)
+
+function rrule(::typeof(getindex), x::T, i::Integer) where {T<:Tuple}
+    function getindex_back_1(dy)
+        dx = ntuple(j -> j == i ? dy : NoTangent(), _tuple_N(T))
+        return (NoTangent(), Tangent{T}(dx...), NoTangent())
+    end
+    return x[i], getindex_back_1
+end
+
+# Special case for tuples of only numbers
+function rrule(::typeof(getindex), x::T, i::Integer) where {T<:NTuple{<:Any,<:Number}}
+    function getindex_back_2(dy_raw)
+        dy = unthunk(dy_raw)
+        dx = ntuple(j -> j == i ? dy : zero(dy), _tuple_N(T))
+        return (NoTangent(), Tangent{T}(dx...), NoTangent())
+    end
+    return x[i], getindex_back_2
+end
+
+function rrule(::typeof(getindex), x::T, inds) where {T<:Tuple}  # e.g. ranges, not type-stable
+    function getindex_back_3(dy_raw)
+        dy = unthunk(dy_raw)
+        dx = ntuple(Returns(NoTangent()), _tuple_N(T))
+        for (dyi, i) in zip(dy, inds)
+            dx = Base.setindex(dx, dyi + dx[i], i)
+        end
+        return (NoTangent(), Tangent{T}(dx...), NoTangent())
```

Comment:
FWIW, Zygote has two rules here, one for `inds:: AbstractUnitRange`, and a generic one:  https://github.com/FluxML/Zygote.jl/blob/master/src/lib/lib.jl#L125-L142  The generic one accumulates within each tuple element, rather than adding the tuples.

I have not compared carefully whether that performs better in some cases, etc.